packages/cmr-client/src/EarthdataLogin.ts

```diff
@@ -1,202 +1,170 @@
   }
+};
 
-  /**
-   * This helper method is called in the functions that retrieve, create, and revoke EarthdataLogin
-   * tokens for error-handling. If API call made to the EarthdataLogin endpoint results in an error.
-   * The statuscode, statusmessage, error description, and error message are thrown and outputted.
-   *
-   * @throws {Error} - EarthdataLogin error
-   */
-  handleHttpError(error: Error) {
-    const statusCode = get(error, 'response.statusCode', error.code);
-    const statusMessage = get(error, 'response.statusMessage', error.message);
-    const responseErrorDescription = JSON.parse(get(error, 'response.body')).error_description;
-    const errorMessage = `EarthdataLogin error: ${responseErrorDescription},  statusCode: ${statusCode}, statusMessage: ${statusMessage}. Earthdata Login Request failed`;
+/**
+ * This helper method is called in the functions that retrieve, create, and revoke EarthdataLogin
+ * tokens for error-handling. If API call made to the EarthdataLogin endpoint results in an error.
+ * The statuscode, statusmessage, error description, and error message are thrown and outputted.
+ *
+ * @returns {Error} - EarthdataLogin error
+ */
+export const parseHttpError = (error: HTTPError, requestType: string): Error => {
+  const statusCode = error.response.statusCode;
+  const statusMessage = error.response.statusMessage || 'Unknown';
+  const errorBody = requestType === 'retrieve' || requestType === 'create' ? JSON.stringify(error.response.body) : error.response.body;
+  const message = `EarthdataLogin error: ${errorBody},  statusCode: ${statusCode}, statusMessage: ${statusMessage}. Earthdata Login Request failed`;
+  return new Error(message);
+};
 
-    throw new Error(errorMessage);
+/**
+ * The method for getting the token from the Earthdata Login endpoint. Sends a GET request
+ * with the users' base64 encoded username and password as a header for authorization. If the
+ * users' credentials are accepted the first unexpired token is retrieved, if one exists, and
+ * returned on their behalf, if not, an error is thrown. If the user does not have a token
+ * in Earthdata Login then undefined is returned to indicate to token needs to be created.
+ *
+ */
+export const retrieveEDLToken = async (
+  username: string,
+  password: string,
+  edlEnv: string
+): Promise<string | undefined> => {
+  // response: get a token from the Earthdata login endpoint using credentials if exists
+  let rawResponse: Response<unknown>;
+  try {
+    rawResponse = await got.get(`${getEdlUrl(edlEnv)}/api/users/tokens`,
+      {
+        responseType: 'json',
+        username,
+        password,
+      });
+  } catch (error) {
+    if (error instanceof got.HTTPError) throw parseHttpError(error, 'retrieve');
+    throw parseCaughtError(error);
   }
+  const tokens = GetTokenResponseBody.parse(rawResponse.body);
+  const isTokenExpired = (token: Token) => (returnJWTexp(token.access_token) < +new Date() / 1000);
```

Comment:
move this method outside as a standalone method, and add doc.
what's this `+` in `+new Date() / 1000`

Reply:
You can use [Date.now()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now)